<h1 style="
color:white;
background-color:#1f77b4;
padding:12px;
border-radius:10px;
text-align:center;">
DATA ANALYTICS PROJECT
</h1>

In [34]:
import pandas as pd
import numpy as np

In [67]:
df=pd.read_excel(r'Dataset for Data Analytics.xlsx')

In [68]:
df


,OrderID,Date,CustomerID,Product,Quantity,UnitPrice,ShippingAddress,PaymentMethod,OrderStatus,TrackingNumber,ItemsInCart,CouponCode,ReferralSource,TotalPrice
0,ORD200000,2023-01-04,C72649,Monitor,5,570.62,928 Main St,Debit Card,Shipped,TRK37947903,7,SAVE10,Instagram,2853.10
1,ORD200001,2024-08-23,C75739,Phone,2,151.35,823 Main St,Online,Shipped,TRK91186779,3,SAVE10,Referral,302.70
2,ORD200002,2024-02-27,C81728,Tablet,5,550.68,512 Main St,Credit Card,Cancelled,TRK42903982,8,FREESHIP,Email,2753.40
3,ORD200003,2023-10-15,C33540,Chair,1,273.19,275 Main St,Debit Card,Returned,TRK62788070,5,SAVE10,Facebook,273.19
4,ORD200004,2025-05-08,C81840,Printer,4,626.01,668 Main St,Online,Delivered,TRK29241424,8,SAVE10,Email,2504.04
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
1195,ORD201195,2024-06-20,C21126,Desk,1,107.04,392 Main St,Credit Card,Cancelled,TRK38009181,6,FREESHIP,Google,107.04
1196,ORD201196,2024-03-04,C20095,Monitor,2,662.53,778 Main St,Online,Cancelled,TRK69207593,5,NaN,Facebook,1325.06
1197,ORD201197,2023-07-13,C79674,Tablet,2,436.84,275 Main St,Online,Delivered,TRK88039356,2,FREESHIP,Instagram,873.68
1198,ORD201198,2024-08-22,C64753,Chair,4,262.52,509 Main St,Debit Card,Cancelled,TRK71683331,4,WINTER15,Instagram,1050.08


In [69]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1200 entries, 0 to 1199
Data columns (total 14 columns):
 #   Column           Non-Null Count  Dtype         
---  ------           --------------  -----         
 0   OrderID          1200 non-null   object        
 1   Date             1200 non-null   datetime64[ns]
 2   CustomerID       1200 non-null   object        
 3   Product          1200 non-null   object        
 4   Quantity         1200 non-null   int64         
 5   UnitPrice        1200 non-null   float64       
 6   ShippingAddress  1200 non-null   object        
 7   PaymentMethod    1200 non-null   object        
 8   OrderStatus      1200 non-null   object        
 9   TrackingNumber   1200 non-null   object        
 10  ItemsInCart      1200 non-null   int64         
 11  CouponCode       891 non-null    object        
 12  ReferralSource   1200 non-null   object        
 13  TotalPrice       1200 non-null   float64       
dtypes: datetime64[ns](1), float64(2), int64(

checking if we have null values

In [27]:
df.isnull().sum()

OrderID              0
Date                 0
CustomerID           0
Product              0
Quantity             0
UnitPrice            0
ShippingAddress      0
PaymentMethod        0
OrderStatus          0
TrackingNumber       0
ItemsInCart          0
CouponCode         309
ReferralSource       0
TotalPrice           0
dtype: int64

since we have null values in couponcode column which mean some customers haven't availed discounts .so we will not drop null values instead we replace them 

In [31]:
df['CouponCode'] = df['CouponCode'].fillna('No Coupon')

In [32]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1200 entries, 0 to 1199
Data columns (total 14 columns):
 #   Column           Non-Null Count  Dtype         
---  ------           --------------  -----         
 0   OrderID          1200 non-null   object        
 1   Date             1200 non-null   datetime64[ns]
 2   CustomerID       1200 non-null   object        
 3   Product          1200 non-null   object        
 4   Quantity         1200 non-null   int64         
 5   UnitPrice        1200 non-null   float64       
 6   ShippingAddress  1200 non-null   object        
 7   PaymentMethod    1200 non-null   object        
 8   OrderStatus      1200 non-null   object        
 9   TrackingNumber   1200 non-null   object        
 10  ItemsInCart      1200 non-null   int64         
 11  CouponCode       1200 non-null   object        
 12  ReferralSource   1200 non-null   object        
 13  TotalPrice       1200 non-null   float64       
dtypes: datetime64[ns](1), float64(2), int64(

checking if quantity and unitPrice are equals to total price 

In [87]:
np.isclose(df['Quantity'] * df['UnitPrice'], df['TotalPrice']).all()

np.True_

yes quantity and unitPrice are equals to total price


In [38]:
df.shape

(1200, 14)

# Sales Analysis

In [40]:
np.sum(df['TotalPrice'])


np.float64(1264761.96)

In [42]:
np.mean(df['TotalPrice'])

np.float64(1053.9683)

# Product Analysis

In [44]:
df['Product'].value_counts()

Product
Printer    181
Tablet     179
Chair      178
Laptop     173
Desk       170
Monitor    163
Phone      156
Name: count, dtype: int64

# Customer Analysis

In [47]:
df['CustomerID'].nunique()

1189

<h1 style=color:blue; padding:10px;">Group Analysis
</h1>

### 1.sales by payment method

In [49]:
df.groupby('PaymentMethod')['TotalPrice'].sum()

PaymentMethod
Cash           259786.29
Credit Card    263847.63
Debit Card     232361.18
Gift Card      246323.92
Online         262442.94
Name: TotalPrice, dtype: float64

### percentage wise

In [81]:
np.divide( 263847.63,1264761.96)*100

np.float64(20.540330767063868)

In [82]:
np.divide(262442.94,1264761.96)*100

np.float64(20.750382150962228)

In [83]:
np.divide(259786.29,1264761.96)*100

np.float64(20.540330767063868)

<div style="font-family: 'Segoe UI', sans-serif; line-height:1.6; margin-top:10px;">

<h3 style="color:#2E86C1; border-bottom:2px solid #2E86C1; padding-bottom:5px;">
💳 Payment Method Analysis
</h3>

<div style="display:flex; gap:10px; flex-wrap:wrap; margin:10px 0;">

  <div style="flex:1; min-width:180px; background:#E8F6F3; padding:12px; border-radius:10px; border-left:5px solid #1ABC9C;">
    <h4 style="margin:0; color:#117864;">Credit Card</h4>
    <p style="font-size:18px; margin:5px 0;"><b>20.86%</b></p>
  </div>

  <div style="flex:1; min-width:180px; background:#FEF9E7; padding:12px; border-radius:10px; border-left:5px solid #F1C40F;">
    <h4 style="margin:0; color:#B7950B;">Online</h4>
    <p style="font-size:18px; margin:5px 0;"><b>20.75%</b></p>
  </div>

  <div style="flex:1; min-width:180px; background:#FDEDEC; padding:12px; border-radius:10px; border-left:5px solid #E74C3C;">
    <h4 style="margin:0; color:#C0392B;">Cash</h4>
    <p style="font-size:18px; margin:5px 0;"><b>20.54%</b></p>
  </div>

</div>

<div style="background:#F4F6F7; padding:15px; border-radius:10px; border-left:5px solid #2E86C1;">

<p style="margin:0;">
<b>Insight:</b> Credit Card payments slightly lead overall transaction value, indicating a strong preference for card-based purchasing behavior. 
Online payments closely follow, reflecting rapid digital adoption across customers. Cash transactions remain nearly equal in share, highlighting that traditional payment methods are still widely used. 

</p>

<p style="margin-top:8px;">
Overall, the distribution is highly balanced, suggesting a <b>diversified payment ecosystem</b> rather than dominance of a single payment method.
</p>

</div>

</div>

### 2.sales by product

In [85]:
df.groupby('Product')['TotalPrice'].sum()

Product
Chair      195620.11
Desk       167459.93
Laptop     192126.56
Monitor    175651.41
Phone      151722.39
Printer    195612.61
Tablet     186568.95
Name: TotalPrice, dtype: float64

<div style="font-family: 'Segoe UI', sans-serif; line-height:1.6;">

<h2 style="color:#2E86C1;">📊 Strategic Business Observation</h2>

<p>The dataset indicates <b>strong revenue concentration</b> in key product segments, particularly office-related and portable technology categories.</p>

<div style="background-color:#E8F6F3; padding:12px; border-left:5px solid #1ABC9C; margin:10px 0;">
<b>Office Furniture & Office Equipment</b><br>
These segments, especially <span style="color:#1F618D;"><b>Chairs</b></span> and <span style="color:#1F618D;"><b>Printers</b></span>, generated the highest revenue contributions, highlighting consistent demand for workplace essentials driven by corporate and institutional usage.
</div>

<div style="background-color:#FEF9E7; padding:12px; border-left:5px solid #F1C40F; margin:10px 0;">
<b>Portable Electronics</b><br>
<span style="color:#D35400;"><b>Laptops</b></span> and <span style="color:#D35400;"><b>Tablets</b></span> also performed strongly, reflecting growing adoption of mobile and flexible computing devices across professional and personal segments.
</div>

<div style="background-color:#FDEDEC; padding:12px; border-left:5px solid #E74C3C; margin:10px 0;">
<b>Underperforming Segment</b><br>
The <span style="color:#C0392B;"><b>Phone category</b></span> contributed the lowest revenue, indicating comparatively weaker demand or stronger market competition within this segment.
</div>

<p style="margin-top:10px;"><i>Overall, the dataset highlights a clear dominance of office-centric and productivity-driven product categories.</i></p>

</div>

# Month-based Analysis

In [86]:
df.groupby(df['Date'].dt.month)['TotalPrice'].sum()

Date
1     124313.23
2     112344.78
3     123840.93
4     109186.05
5     135142.59
6     170616.13
7      85784.64
8      86343.21
9      69321.65
10     89834.82
11     75493.43
12     82540.50
Name: TotalPrice, dtype: float64

# Discount / Coupon Impact

In [90]:
df.groupby('CouponCode')['TotalPrice'].mean()

CouponCode
FREESHIP    1070.405719
SAVE10      1065.874196
WINTER15    1035.902534
Name: TotalPrice, dtype: float64

<div style="font-family: 'Segoe UI', sans-serif; line-height:1.6; margin-top:10px;">

<h3 style="color:#2E86C1; border-bottom:2px solid #2E86C1; padding-bottom:5px;">
🏷️ Promotional Coupon Analysis
</h3>

<div style="display:flex; gap:10px; flex-wrap:wrap; margin:12px 0;">

  <div style="flex:1; min-width:180px; background:#E8F6F3; padding:12px; border-radius:10px; border-left:5px solid #1ABC9C;">
    <h4 style="margin:0; color:#117864;">FREESHIP</h4>
    <p style="font-size:18px; margin:5px 0;"><b>1070.41</b></p>
  </div>

  <div style="flex:1; min-width:180px; background:#FEF9E7; padding:12px; border-radius:10px; border-left:5px solid #F1C40F;">
    <h4 style="margin:0; color:#B7950B;">SAVE10</h4>
    <p style="font-size:18px; margin:5px 0;"><b>1065.87</b></p>
  </div>

  <div style="flex:1; min-width:180px; background:#FDEDEC; padding:12px; border-radius:10px; border-left:5px solid #E74C3C;">
    <h4 style="margin:0; color:#C0392B;">WINTER15</h4>
    <p style="font-size:18px; margin:5px 0;"><b>1035.90</b></p>
  </div>

</div>

<div style="background:#F4F6F7; padding:15px; border-radius:10px; border-left:5px solid #2E86C1;">

<p style="margin:0;">
<b>Insight:</b> The <span style="color:#117864;"><b>FREESHIP</b></span> promotional campaign generated the highest overall contribution, indicating that customers are highly influenced by shipping-related incentives. 
<span style="color:#B7950B;"><b>SAVE10</b></span> performed nearly equally well, suggesting that direct discount-based offers also strongly encourage purchasing behavior.
</p>

<p style="margin-top:8px;">
Meanwhile, <span style="color:#C0392B;"><b>WINTER15</b></span> showed slightly lower performance compared to the other campaigns, though the difference remains relatively small. This indicates that all promotional strategies were effective, with customer response distributed fairly evenly across campaigns.
</p>

<p style="margin-top:8px;">
Overall, the analysis suggests that both <b>free shipping incentives</b> and <b>percentage-based discounts</b> play a significant role in driving customer engagement and sales conversions.
</p>

</div>

</div>

# Filtering High value orders

In [93]:
df[df['TotalPrice']>1000]

,OrderID,Date,CustomerID,Product,Quantity,UnitPrice,ShippingAddress,PaymentMethod,OrderStatus,TrackingNumber,ItemsInCart,CouponCode,ReferralSource,TotalPrice
0,ORD200000,2023-01-04,C72649,Monitor,5,570.62,928 Main St,Debit Card,Shipped,TRK37947903,7,SAVE10,Instagram,2853.10
2,ORD200002,2024-02-27,C81728,Tablet,5,550.68,512 Main St,Credit Card,Cancelled,TRK42903982,8,FREESHIP,Email,2753.40
4,ORD200004,2025-05-08,C81840,Printer,4,626.01,668 Main St,Online,Delivered,TRK29241424,8,SAVE10,Email,2504.04
9,ORD200009,2023-11-21,C31946,Desk,4,509.38,102 Main St,Credit Card,Shipped,TRK33478363,6,SAVE10,Google,2037.52
10,ORD200010,2023-12-29,C43443,Tablet,5,625.97,333 Main St,Credit Card,Returned,TRK98859248,9,WINTER15,Instagram,3129.85
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
1191,ORD201191,2025-02-10,C44148,Desk,4,376.43,809 Main St,Gift Card,Pending,TRK67872709,5,WINTER15,Instagram,1505.72
1192,ORD201192,2024-02-28,C72007,Chair,3,432.85,999 Main St,Online,Pending,TRK16240916,4,NaN,Instagram,1298.55
1196,ORD201196,2024-03-04,C20095,Monitor,2,662.53,778 Main St,Online,Cancelled,TRK69207593,5,NaN,Facebook,1325.06
1198,ORD201198,2024-08-22,C64753,Chair,4,262.52,509 Main St,Debit Card,Cancelled,TRK71683331,4,WINTER15,Instagram,1050.08


# Correlation Analysis

In [94]:
df[['Quantity','UnitPrice','TotalPrice']].corr()

,Quantity,UnitPrice,TotalPrice
Quantity,1.000000,0.014553,0.615251
UnitPrice,0.014553,1.000000,0.717081
TotalPrice,0.615251,0.717081,1.000000


<div style="font-family:'Segoe UI', sans-serif; line-height:1.6;">

<h3 style="color:#2E86C1; border-bottom:2px solid #2E86C1; padding-bottom:5px;">
📊 Correlation Analysis
</h3>

<div style="background:#F4F6F7; padding:15px; border-radius:10px; border-left:5px solid #2E86C1;">

<p>
The correlation analysis reveals a strong positive relationship between 
<b>UnitPrice</b> and <b>TotalPrice</b> (<b>0.717</b>), indicating that higher-priced products contribute significantly to overall revenue generation.
</p>

<p>
A moderately positive correlation exists between 
<b>Quantity</b> and <b>TotalPrice</b> (<b>0.615</b>), suggesting that larger purchase volumes also play an important role in increasing transaction value.
</p>

<p>
In contrast, the correlation between 
<b>Quantity</b> and <b>UnitPrice</b> (<b>0.014</b>) is extremely weak, indicating that product pricing has minimal influence on the quantity purchased by customers.
</p>

<p>
Overall, the analysis suggests that both product pricing and purchase volume contribute to revenue growth, with pricing showing a comparatively stronger impact.
</p>

</div>

</div>

# product status analysis


In [98]:
df.groupby(["Product",'OrderStatus']).size().unstack()

OrderStatus,Cancelled,Delivered,Pending,Returned,Shipped
Product,,,,,
Chair,45,33,41,28,31
Desk,35,32,38,32,33
Laptop,35,40,30,39,29
Monitor,35,31,29,36,32
Phone,31,38,28,31,28
Printer,35,29,37,38,42
Tablet,34,28,34,43,40


<div style="font-family:'Segoe UI', sans-serif; line-height:1.7; margin-top:10px;">

<h3 style="color:#2E86C1; border-bottom:2px solid #2E86C1; padding-bottom:5px;">
📦 Product-wise Order Status Analysis
</h3>

<div style="background:#F4F6F7; padding:15px; border-radius:10px; border-left:5px solid #2E86C1;">

<p>
The order status analysis reveals distinct operational and customer behavior patterns across different product categories.
</p>

<p>
<b>Chair</b> products recorded the highest number of <span style="color:#C0392B;"><b>Cancelled orders (45)</b></span>, indicating potential issues related to pricing, delivery expectations, or purchase reconsideration within the furniture category.
</p>

<p>
<b>Printer</b> and <b>Tablet</b> categories demonstrated strong logistics performance, generating the highest number of <span style="color:#117864;"><b>Shipped orders</b></span> with values of <b>42</b> and <b>40</b> respectively. This suggests strong fulfillment efficiency and stable customer demand for these products.
</p>

<p>
<b>Laptops</b> achieved the highest number of <span style="color:#1F618D;"><b>Delivered orders (40)</b></span>, reflecting successful order completion and strong customer acceptance within the portable electronics segment.
</p>

<p>
However, <b>Tablets</b> also experienced the highest number of <span style="color:#D35400;"><b>Returned orders (43)</b></span>, followed closely by <b>Laptops (39)</b> and <b>Printers (38)</b>. This may indicate product expectation mismatches, quality concerns, or post-purchase dissatisfaction in technology-related categories.
</p>

<p>
Across most products, <b>Pending orders</b> remained relatively balanced, suggesting a steady order processing pipeline without extreme operational bottlenecks.
</p>

<p>
Overall, the analysis indicates that while electronics and office-related products generate strong shipping and delivery activity, they also face comparatively higher return rates, highlighting the importance of improving product quality assurance, customer expectations, and after-sales support.
</p>

</div>

</div>



# ReferralSource analysis

In [99]:
df.groupby(["Product",'ReferralSource']).size().unstack()

ReferralSource,Email,Facebook,Google,Instagram,Referral
Product,,,,,
Chair,41,37,28,41,31
Desk,29,32,39,41,29
Laptop,33,31,33,40,36
Monitor,32,37,26,39,29
Phone,34,24,39,26,33
Printer,41,36,41,31,32
Tablet,40,31,35,41,32


<div style="font-family:'Segoe UI', sans-serif; line-height:1.7; margin-top:10px;">

<h3 style="color:#2E86C1; border-bottom:2px solid #2E86C1; padding-bottom:5px;">
🌐 Referral Source Performance Analysis
</h3>

<div style="background:#F4F6F7; padding:15px; border-radius:10px; border-left:5px solid #2E86C1;">

<p>
The referral source analysis highlights the effectiveness of different marketing and customer acquisition channels across product categories.
</p>

<p>
<span style="color:#8E44AD;"><b>Instagram</b></span> emerged as one of the strongest traffic-generating platforms, contributing high engagement across multiple products, particularly for <b>Chair</b>, <b>Desk</b>, and <b>Tablet</b> categories. This suggests strong customer interaction and product visibility through visual and social-media-driven marketing campaigns.
</p>

<p>
<span style="color:#117864;"><b>Email marketing</b></span> also demonstrated strong performance, especially for <b>Chair</b>, <b>Printer</b>, and <b>Tablet</b> products, indicating that direct customer communication remains an effective conversion channel.
</p>

<p>
<span style="color:#D35400;"><b>Google referrals</b></span> contributed significantly to products such as <b>Desk</b>, <b>Phone</b>, and <b>Printer</b>, highlighting the importance of search visibility and customer intent-driven traffic in generating product interest.
</p>

<p>
In contrast, <span style="color:#C0392B;"><b>Facebook</b></span> showed comparatively lower engagement for certain categories, particularly <b>Phone</b> products, suggesting reduced effectiveness or lower audience responsiveness within that segment.
</p>

<p>
Additionally, <b>Referral-based traffic</b> remained relatively balanced across products, indicating a stable contribution from customer recommendations and external references.
</p>

<p>
Overall, the analysis suggests that <b>Instagram, Email campaigns, and Google search traffic</b> are the most influential acquisition channels within the dataset, emphasizing the importance of digital marketing strategies and targeted online engagement.
</p>

</div>

</div>